pkg/services/elasticstore/events.go

```diff
@@ -42,7 +47,33 @@ func GetEventsQuery(query *m.GetEventsQuery) error {
 			},
 		},
 	}
-	out, err := es.Search("events", "", map[string]interface{}{"size": query.Size, "sort": "timestamp:desc"}, esQuery)
+	var out elastigo.SearchResult
+	var err error
+
+	wildcard := setting.Cfg.Section("elasticsearch").Key("wildcard_events").MustBool()
+
+	if wildcard {
+		out, err = es.Search("events*", "", map[string]interface{}{"size": query.Size, "sort": "timestamp:desc"}, esQuery)
+	} else {
+		start := time.Unix(query.Start/1000, 0)
+		end := time.Unix(query.End/1000, 0)
+		r := end.Sub(start) / time.Hour / 24
+		idxDates := make([]string, 0, r+1)
+		y, m, d := start.Date()
+		if r > 0 {
+			for {
+				end = end.Add(-(time.Hour * 24))
+				y2, m2, d2 := end.Date()
+				idxDates = append(idxDates, fmt.Sprintf("events-%d-%02d-%02d", y, m, d))
+				if y2 <= y && m2 <= m && d2 <= d {
+					break
+				}
+			}
+		}
+		allTogether := strings.Join(idxDates, ",")
```

Comment:
When testing this, i just get an empty string resulting in all indexes, including the "metric" index being searched.